tools/src/wyvern/target/corewyvernIL/support/VarGenContext.java

```diff
@@ -30,8 +40,13 @@ public String endToString() {
 	public Expression lookupExp(String varName) {
 		if (varName.equals(var))
 			return expr;
-		else
+		else if (genContext.getType(varName) != null) {
```

Comment:
This case of the if needs an explanation in a comment, I think...why are we testing for the type being defined when the person is looking for a variable?